src/distances.jl

```diff
@@ -19,13 +18,31 @@ ITensors.inner(ρ::MPO, σ::LPDO{MPS}) =
   inner(σ.X, ρ, σ.X)
 
 """
-    fidelity(ψ::MPS, ϕ::MPS)
+    fidelity(ρ::ITensor, σ::ITensor)
 
-Compute the fidelity between two MPS:
+Compute the quantum fidelity between two ITensors, which are treated as density operators
+from the unprimed to the primed indices (if they are matrix-like).
+
+Matrix-like ITensors should be Hermitian and non-negative.
+"""
+function fidelity(ρ::ITensor{N}, σ::ITensor{N}; warnings::Bool = true) where {N}
```

Comment:
Another way to design the warning would be to control it with a global setting and macros, like the `@disable_warn_order` ITensors.jl macro: https://github.com/ITensor/ITensors.jl/blob/db7a2961415f96b42d1ec57a4dae02bb56ab4bcf/src/global_variables.jl#L2-L130

We could call it `@enable_full_fidelity_warning` and `@disable_full_fidelity_warning`.

Maybe the keyword argument should have a more verbose name, like `enable_full_fidelity_warning`? Otherwise it makes it sounds like it turns off all warnings, which may not be the case.

Also, is the point of this warning that someone may accidentally call this function when using something like `fidelity(::MPO, ::MPO)`? If that is the case, it seems better to turn this off by default in `fidelity(::ITensor, ::ITensor)`, but then have it on by default in `fidelity(::MPO, ::MPO)` and related functions. If someone is calling it directly with `fidelity(::ITensor, ::ITensor)` I think it is clear that it is not an efficient operation.

Reply:
#183